Add check to confirm remote configuration files match

nitisht opened the pull request:
## Description
Currently it is possible to run distributed Minio server instances
each with its own configuration file (having different settings
enabled).

For example one Minio instance in a distributed setup may
have event notifications enabled, while others don't. This doesn't
cause any functional issues, and only the instance with notifications
enabled will actually send out event notifications.

Such situation can however lead users to believe that all instances
will work in same manner. This PR adds a check to make sure MD5sum
of each instance's configuration files match, hence making sure all
instances are running same configuration file.

## Motivation and Context
Observed after addition of storage classes

## How Has This Been Tested?
Manually by running distributed Minio instances with different config files

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

krisis commented on cmd/config-current.go:
```diff
@@ -138,6 +141,20 @@ func (s *serverConfig) GetStorageClass() (storageClass, storageClass) {
 	return s.StorageClass.Standard, s.StorageClass.RRS
 }
 
+// Calculates MD5 hash of the config file
```
we should compute the config hash only on startup and after an `mc admin config set ...` operation. For every RPC call, we should return the hash computed previously.

nitisht committed:
Add check to confirm if remote configuration files match

Currently it is possible to run distributed Minio server instances
each with its own configuration file (having different settings
enabled).

For example one Minio instance in a distributed setup may
have event notifications enabled, while others don't. This doesn't
cause any functional issues, and only the instance with notifications
enabled will actually send out event notifications.

Such situation can however lead users to believe that all instances
will work in same manner. This PR adds a check to make sure MD5sum
of each instance's configuration files match, hence making sure all
instances are running same configuration file.
